dso-l2/src/main/java/com/tc/l2/logging/TCLogbackLogging.java

```diff
@@ -0,0 +1,153 @@
+package com.tc.l2.logging;
+
+import ch.qos.logback.classic.LoggerContext;
+import ch.qos.logback.classic.spi.ILoggingEvent;
+import ch.qos.logback.core.Appender;
+import ch.qos.logback.core.rolling.FixedWindowRollingPolicy;
+import ch.qos.logback.core.rolling.RollingFileAppender;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.tc.logging.TCLogging;
+import com.tc.util.ProductInfo;
+
+import java.lang.management.ManagementFactory;
+import java.lang.management.RuntimeMXBean;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.Properties;
+
+public class TCLogbackLogging {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(TCLogbackLogging.class);
+  private static final Logger CONSOLE_LOGGER = TCLogging.getConsoleLogger();
+
+  public static void initLogging() {
+    writeVersion();
+    writePID();
+  }
+
+  public static void redirectLogging(String logDir) {
+    LoggerContext loggerContext = (LoggerContext) LoggerFactory.getILoggerFactory();
+    ch.qos.logback.classic.Logger internalLogger = loggerContext.getLogger(Logger.ROOT_LOGGER_NAME);
+    Appender<ILoggingEvent> appender = internalLogger.getAppender("TC_BASE");
+    if (appender != null) {
+      String logLocation = logDir + "/terracotta.server.log";
+      LOGGER.info("Log file: {}", logLocation);
+      BufferingAppender<ILoggingEvent> base = (BufferingAppender) appender;
```

Comment:
Currently I'm relying on a special appender named "TC_BASE" which is of type `BufferingAppender`. But nothing is stopping a user from overriding with his config having the same name. So yeah I guess it needs to be handled better to be on the safer side.